File: solutions/lua/armstrong-numbers/1/armstrong-numbers.lua
local ArmstrongNumbers = {}

function sumOfCubes(n)
    local sum = 0
    local nStr = tostring(n)
    for i = 1, #nStr do
        local digit = tonumber(nStr:sub(i, i))
        sum = sum + digit ^ 3
    end
    return sum
end

function ArmstrongNumbers.is_armstrong_number(number)
    local num = sumOfCubes(number)
    if num == number then
        return true
    else
        return false
    end
end

return ArmstrongNumbers

File: solutions/nim/armstrong-numbers/1/armstrong_numbers.nim
import sequtils
import parseutils
import math
proc isArmstrongNumber*(n: int): bool =
  # convert to string
  var str = $n
  # get length of that string
  var exp = str.len()
  # split string into digit chars
  var arr = str.toSeq()
  # initialize a seq of ints
  var intarr = newSeq[int](0)

  for c in arr:
    var res: int
    # parse char into res variable
    discard parseInt($c, res, 0)
    # calculate the exponent
    var toPow = pow(float64(res), float64(exp))
    # round the result of pow to int
    res = int(round(toPow))
    # add the result to array
    intarr.add(res)
  # sum the array (similar to reduce)
  var folded = foldl(intarr, a+b)
  # check if sum is equal to the initial number
  return folded == n
